PHPStan check fails on PHP 8

```
$ php -v
PHP 8.0.2 (cli) (built: Feb 28 2021 17:15:21) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.2, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.2, Copyright (c), by Zend Technologies
    with Xdebug v3.0.3, Copyright (c) 2002-2021, by Derick Rethans

$ git rev-parse HEAD
fb1658ff8ab3f6d7880fb185d66c806215f87f28

$ phpstan --version
PHPStan - PHP Static Analysis Tool 0.12.55

$ phpstan a
 306/306 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ---------------------------------------------------------------------------------------
  Line   Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php
 ------ ---------------------------------------------------------------------------------------
  72     Property Doctrine\DBAL\Driver\Mysqli\MysqliConnection::$conn (mysqli) does not accept
         mysqli|false.
 ------ ---------------------------------------------------------------------------------------

 ------ ------------------------------------------------------------------------------------
  Line   Doctrine/DBAL/Driver/PDOConnection.php
 ------ ------------------------------------------------------------------------------------
  66     Method Doctrine\DBAL\Driver\PDOConnection::getServerVersion() should return string
         but returns array|bool|int|string|null.
  102    Method Doctrine\DBAL\Driver\PDOConnection::lastInsertId() should return string but
         returns string|false.
  105    Method Doctrine\DBAL\Driver\PDOConnection::lastInsertId() should return string but
         returns string|false.
 ------ ------------------------------------------------------------------------------------

 ------ --------------------------------------------------------------------------
  Line   Doctrine/DBAL/DriverManager.php
 ------ --------------------------------------------------------------------------
  260    Parameter #2 $haystack of function in_array expects array, array<string,
         string>|false given.
 ------ --------------------------------------------------------------------------

 ------ -----------------------------------------------------
  Line   Doctrine/DBAL/Tools/Dumper.php
 ------ -----------------------------------------------------
  119    Cannot call method getName() on DateTimeZone|false.
 ------ -----------------------------------------------------

 [ERROR] Found 6 errors

$ composer require --dev phpstan/phpstan '^0.12'
./composer.json has been updated
Running composer update phpstan/phpstan
Loading composer repositories with package information
Updating dependencies
Lock file operations: 0 installs, 1 update, 0 removals
  - Upgrading phpstan/phpstan (0.12.55 => 0.12.80)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 0 installs, 1 update, 0 removals
  - Upgrading phpstan/phpstan (0.12.55 => 0.12.80): Extracting archive
Generating autoload files
composer/package-versions-deprecated: Generating version class...
composer/package-versions-deprecated: ...done generating version class
33 packages you are using are looking for funding.
Use the `composer fund` command to find out more!

$ phpstan a
 306/306 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ---------------------------------------------------------------------------------------
  Line   Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php
 ------ ---------------------------------------------------------------------------------------
  72     Property Doctrine\DBAL\Driver\Mysqli\MysqliConnection::$conn (mysqli) does not accept
         mysqli|false.
 ------ ---------------------------------------------------------------------------------------

 ------ --------------------------------------------------------------------------
  Line   Doctrine/DBAL/DriverManager.php
 ------ --------------------------------------------------------------------------
  260    Parameter #2 $haystack of function in_array expects array, array<string,
         string>|false given.
 ------ --------------------------------------------------------------------------

 [ERROR] Found 2 errors
```

Looks like most of the issues will be resolved by a PHPStan upgrade but the rest need some attention.

--- a/composer.json
+++ b/composer.json
@@ -40,7 +40,7 @@
     "require-dev": {
         "doctrine/coding-standard": "8.2.0",
         "jetbrains/phpstorm-stubs": "2019.3",
-        "phpstan/phpstan": "0.12.55",
+        "phpstan/phpstan": "0.12.80",
         "phpunit/phpunit": "9.5.0",
         "psalm/plugin-phpunit": "0.13.0",
         "symfony/console": "^2.0.5|^3.0|^4.0|^5.0",

--- a/lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php
+++ b/lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php
@@ -11,6 +11,7 @@ use Doctrine\DBAL\Driver\ServerInfoAwareConnection;
 use Doctrine\DBAL\ParameterType;
 use mysqli;
 
+use function assert;
 use function floor;
 use function func_get_args;
 use function in_array;
@@ -69,7 +70,10 @@ class MysqliConnection implements ConnectionInterface, PingableConnection, Serve
 
         $flags = $driverOptions[static::OPTION_FLAGS] ?? null;
 
-        $this->conn = mysqli_init();
+        $conn = mysqli_init();
+        assert($conn !== false);
+
+        $this->conn = $conn;
 
         $this->setSecureConnection($params);
         $this->setDriverOptions($driverOptions);

--- a/lib/Doctrine/DBAL/DriverManager.php
+++ b/lib/Doctrine/DBAL/DriverManager.php
@@ -257,7 +257,9 @@ final class DriverManager
     private static function createDriver(array $params): Driver
     {
         if (isset($params['driverClass'])) {
-            if (! in_array(Driver::class, class_implements($params['driverClass'], true))) {
+            $interfaces = class_implements($params['driverClass'], true);
+
+            if ($interfaces === false || ! in_array(Driver::class, $interfaces)) {
                 throw Exception::invalidDriverClass($params['driverClass']);
             }
 

